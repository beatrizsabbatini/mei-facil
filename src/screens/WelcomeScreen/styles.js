import { StyleSheet } from 'react-native';

import { colors, fonts, metrics } from '../../styles/index';

const styles = StyleSheet.create({
	container: {
		flex: 2,
		backgroundColor: colors.Primary,
		alignItems: 'center',
		justifyContent: 'space-between',
		padding: metrics.bigMargin,
	},
	welcomeTitle: {
		fontFamily: fonts.robotoBold,
		color: colors.White,
		fontSize: fonts.big,
		paddingVertical: metrics.baseMargin,
	},
	welcomeMessage: {
		fontFamily: fonts.robotoRegular,
		fontSize: fonts.big,
		color: colors.White,
		textAlign: 'center',
		width: '80%',
		paddingVertical: metrics.doubleBaseMargin,
	},
	frequentQuestions: {
		fontSize: fonts.regular,
		color: colors.White,
		textTransform: 'uppercase',
		letterSpacing: 2,
		fontFamily: fonts.robotoBold,
		paddingVertical: metrics.doubleBaseMargin,
	},
	basicView: {
		width: '100%',
		flex: 1,
		alignItems: 'center',
	},
	centeredView: {
		justifyContent: 'center',
	},
});

export default styles;
