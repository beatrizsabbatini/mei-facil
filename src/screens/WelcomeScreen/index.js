import { Fontisto } from '@expo/vector-icons';
import * as WebBrowser from 'expo-web-browser';
import React from 'react';
import { Text, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Button from '../../components/Button';
import Logo from '../../components/Logo';
import { colors } from '../../styles';
import styles from './styles';

const WelcomeScreen = () => {
  async function openBrowser() {
    await WebBrowser.openBrowserAsync('https://www.meifacil.com/duvidas');
  }

  return (
    <View style={styles.container}>
      <Logo />
      <View style={[styles.basicView, styles.centeredView]}>
        <Text style={styles.welcomeTitle}>Bem-vindo!</Text>
        <Text style={styles.welcomeMessage}>
          Tudo que você precisa para o seu MEI em um só lugar
        </Text>
      </View>
      <View style={styles.basicView}>
        <Button
          text="Tenho CNPJ mei"
          icon={
            <Fontisto name="arrow-right-l" size={24} color={colors.Black} />
          }
        />
        <Button
          text="Abrir um CNPJ mei"
          icon={
            <Fontisto name="arrow-right-l" size={24} color={colors.Black} />
          }
        />
        <TouchableOpacity onPress={() => openBrowser()}>
          <Text style={styles.frequentQuestions}>perguntas frequentes</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default WelcomeScreen;
