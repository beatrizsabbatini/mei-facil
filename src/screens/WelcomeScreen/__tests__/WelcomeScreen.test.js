// __tests__/Intro-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from '..';

test('renders correctly', () => {
  const tree = renderer.create(<WelcomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
