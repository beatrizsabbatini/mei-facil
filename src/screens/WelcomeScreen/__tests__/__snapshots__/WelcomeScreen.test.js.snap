// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00A5EF",
      "flex": 2,
      "justifyContent": "space-between",
      "padding": 25,
    }
  }
>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": -100,
          },
        ],
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 50,
          "resizeMode": "contain",
          "width": 200,
        }
      }
    />
  </View>
</View>
`;
